backend-tests/backend-tests.hs

```diff
@@ -254,6 +254,21 @@ queryIIReject cid user_id l x = do
     Rejected _ -> return ()
     Replied _ -> lift $ assertFailure "queryIIReject: Unexpected reply"
 
+queryIIRejectWith :: forall s a b.
+  HasCallStack =>
+  KnownSymbol s =>
+  (a -> IO b) ~ (IIInterface IO .! s) =>
+  Candid.CandidArg a =>
+  BS.ByteString -> EntityId -> Label s -> a -> String -> M ()
+queryIIRejectWith cid user_id l x expectedMessagePattern = do
+  r <- submitQuery $ QueryRequest (EntityId cid) user_id (symbolVal l) (Candid.encode x)
+  case r of
+    Rejected (_code, msg) ->
+      if not (msg =~ expectedMessagePattern)
+      then liftIO $ assertFailure $ printf "expected error matching %s, got: %s" (show expectedMessagePattern) (show msg)
+      else return ()
```

Comment:
A bit late, but why not use `unless` instead of `if not … then  … else return ()`?